lib/db/mysql.js

```diff
@@ -304,15 +327,20 @@ module.exports = function (log, error) {
         deviceInfo.callbackPublicKey,
         deviceInfo.callbackAuthKey
       ]
-    )
+    }, ...capabilitiesQueries])
   }
 
   var UPDATE_DEVICE = 'CALL updateDevice_5(?, ?, ?, ?, ?, ?, ?, ?, ?)'
 
   MySql.prototype.updateDevice = function (uid, deviceId, deviceInfo) {
-    return this.write(
-      UPDATE_DEVICE,
-      [
+    const capabilitiesQueries = []
+    if (deviceInfo.hasOwnProperty('capabilities')) {
+      capabilitiesQueries.push({sql: PURGE_CAPABILITIES, params: [uid, deviceId]})
+      addLegalCapabilities(uid, deviceId, deviceInfo, capabilitiesQueries)
```

Comment:
Bike-shedding but I think, on balance, I don't like this pattern of passing in the array and modifying it (or not) inside `addLegalCapabilities`. It's kind of small-scale action-at-a-distance.

Would it be cleaner if `addLegalCapabilities` always returns a fresh array and the caller can then be explicit about joining them with `capabilitiesQueries.concat` if it needs to?